Libplanet/Net/NetMQTransport.cs

```diff
@@ -893,19 +893,26 @@ private async Task CreatePermission(BoundPeer peer)
 
             foreach (IPAddress ip in ips)
             {
-                var ep = new IPEndPoint(ip, peer.EndPoint.Port);
-                if (IPAddress.IsLoopback(ip))
+                try
                 {
-                    // This translation is only used in test case because a
-                    // seed node exposes loopback address as public address to
-                    // other node in test case
-                    ep = await _turnClient.GetMappedAddressAsync(cts.Token);
-                }
+                    var ep = new IPEndPoint(ip, peer.EndPoint.Port);
+                    if (IPAddress.IsLoopback(ip))
+                    {
+                        // This translation is only used in test case because a
+                        // seed node exposes loopback address as public address to
+                        // other node in test case
+                        ep = await _turnClient.GetMappedAddressAsync(cts.Token);
+                    }
 
-                // FIXME Can we really ignore IPv6 case?
-                if (ip.AddressFamily.Equals(AddressFamily.InterNetwork))
+                    // FIXME Can we really ignore IPv6 case?
+                    if (ip.AddressFamily.Equals(AddressFamily.InterNetwork))
+                    {
+                        await _turnClient.CreatePermissionAsync(ep, cts.Token);
+                    }
+                }
+                catch (TaskCanceledException)
                 {
-                    await _turnClient.CreatePermissionAsync(ep, cts.Token);
+                    _logger.Debug($"Task is cancelled during {nameof(CreatePermission)}");
                 }
```

Comment:
How about separately wrapping those two `await` calls with `try`–`catch` and logging what specific task (`GetMappedAddressAsync()` or `CreatePermissionAsync()`) was cancelled?

Reply:
How about adding an exception to log to prevent continuing execution when `GetMappedAddressAsync` is canceled?

Reply:
That would be better! (1. logging what specific task was cancelled 2. logging why [= what exception does make] that task was cancelled).

Reply:
```
[16:29:01 DBG] Timeout occurred during CreatePermission: System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at Libplanet.Stun.TurnClient.SendMessageAsync(NetworkStream stream, StunMessage message, CancellationToken cancellationToken)
   at Libplanet.Stun.TurnClient.CreatePermissionAsync(IPEndPoint peerAddress, CancellationToken cancellationToken)
   at Libplanet.Net.NetMQTransport.CreatePermission(BoundPeer peer)
```
How about like this?

Reply:
- Can the message `Timeout occurred during CreatePermission` be `Timeout occurred during GetMappedAddressAsync` as well?
- How about this: `CreatePermissionAsync() has timed out`?

Reply:
`CreatePermission` is the name of the method name that calls `GetMappedAddressAsync` and `CreatePermissionAsync`. We can check `GetMappedAddressAsync` and `CreatePermissionAsync` in the stack trace.